### ISSUE TITLE
User auth with OAuth

### ISSUE THREAD

**Issue by mctaylorpants** [alextaylor000/hoppist_web_app#12]
At least Twitter.

**Comment by mctaylorpants** [comment]
See the oauth-integration branch for progress. To do:
- Set up the token controller for Twitter user sign-in flow. 
- Research how best to write tests for OAuth.
- Pass signed UID to React using a JWT.

**Comment by mctaylorpants** [comment]
I decided to use [Omniauth](https://github.com/intridea/omniauth) instead, since it has such buttery-smooth integration with most platforms and all the features I need.

I set up Omniauth for developer mode, Twitter, and Google, using the default routes that Omniauth provides. The **omniauth** controller is responsible for handling callbacks. It looks up a user by uid and provider, or creates it if it doesn't exist.

The user's first/last name is filled from the info.name field in the auth hash. To make this work, I made **last_name** optional for users, since there's a chance (especially with Twitter) that there will be no last name. I think we should add a second step after logging in to allow the user to customize their display name.

We could also allow users to connect multiple accounts (I know I always end up forgetting what service I used to log in to stuff like this). To do that we'd need to create another table to allow users to have multiple authentication providers. See issue #17.
